app/src/main/java/com/example/weather/ui/map/MapFragment.kt

```diff
@@ -0,0 +1,289 @@
+                    } else {
+                        Toast.makeText(
+                            context,
+                            getString(R.string.location_not_exist),
+                            Toast.LENGTH_SHORT
+                        ).show()
+                    }
+                }
+                return false
+            }
+
+            override fun onQueryTextChange(newText: String?): Boolean {
+                return false
+            }
+        })
+
+        binding.btnFavoriteLocation.setOnClickListener {
+            // TODO implement later
+        }
+
+        binding.icBack.setOnClickListener {
+            goBackFragment()
+        }
+
+        binding.layoutWeatherMap.cardViewMap.setOnClickListener {
+            activity?.let {
+                mWeather?.let { weather -> DetailFragment.newInstance(weather) }
+                    ?.let { detailFragment ->
+                        (it as AppCompatActivity).replaceFragmentToActivity(
+                            it.supportFragmentManager,
+                            detailFragment,
+                            R.id.container
+                        )
+                    }
+            }
+        }
+
+        binding.layoutWeatherMap.icStar.setOnClickListener {
+            binding.layoutWeatherMap.icStar.isClickable = false
+            if (mWeather?.isFavorite == Constant.FALSE) {
+                binding.layoutWeatherMap.icStar.setImageResource(R.drawable.ic_star_yellow)
+                mWeather?.let { weather -> viewModel.insertFavoriteWeather(weather) }
+                mWeather?.isFavorite = Constant.TRUE
+                binding.layoutWeatherMap.icStar.isClickable = true
+            } else {
+                binding.layoutWeatherMap.icStar.setImageResource(R.drawable.ic_star_white)
+                val id = mWeather?.id
+                if (id != null) {
+                    viewModel.removeFavoriteWeather(id)
+                }
+                mWeather?.isFavorite = Constant.FALSE
+                binding.layoutWeatherMap.icStar.isClickable = true
+            }
+        }
+    }
+
+    @Suppress("NestedBlockDepth")
+    override fun initData() {
+        sharedViewModel.isNetworkAvailable.observe(viewLifecycleOwner) { isNetworkEnable ->
+            mIsNetworkEnable = isNetworkEnable
+        }
+
+        mFusedLocationClient = context?.let { LocationServices.getFusedLocationProviderClient(it) }
+
+        arguments?.let { args ->
+            if (mSelectedLatitude == 0.0 && mSelectedLongitude == 0.0) {
+                mWeather = args.getParcelable(Constant.WEATHER_KEY)
+                mSelectedLatitude = mWeather?.latitude ?: 0.0
+                mSelectedLongitude = mWeather?.longitude ?: 0.0
+            }
+            val dateTime = mWeather?.weatherCurrent?.dateTime
+            if (dateTime?.let { dateTime -> Utils.checkTimeInterval(dateTime) } == true && mIsNetworkEnable) {
```

Comment:
Có lẽ lên viết rõ ràng các điều kiện này và null check thay vì lồng trong biểu thức điều kiện ạ
```suggestion
            if (dateTime?.let { dateTime -> Utils.checkTimeInterval(dateTime) } == true && mIsNetworkEnable) {
```